internal/netxlite/tls.go

```diff
@@ -203,10 +214,19 @@ func (h *tlsHandshakerConfigurable) Handshake(
 	if err != nil {
 		return nil, tls.ConnectionState{}, err
 	}
-	if err := tlsconn.HandshakeContext(ctx); err != nil {
-		return nil, tls.ConnectionState{}, err
+	remoteAddr := conn.RemoteAddr().String()
+	trace := ContextTraceOrDefault(ctx)
+	started := time.Now()
+	trace.OnTLSHandshakeStart(remoteAddr, config)
+	err = tlsconn.HandshakeContext(ctx)
+	err = maybeNewErrWrapper(classifyTLSHandshakeError, TLSHandshakeOperation, err)
+	finished := time.Now()
+	state := tlsMaybeConnectionState(tlsconn, err)
+	trace.OnTLSHandshakeDone(started, remoteAddr, config, state, err, finished)
+	if err != nil {
+		return nil, state, err
```

Comment:
We should test that the state we return here is always the zero state.

Reply:
This is now tracked here: https://github.com/ooni/probe-cli/pull/815#pullrequestreview-1015249436